src/engine/mod.rs

```diff
@@ -564,6 +564,36 @@ where
         Ok(None)
     }
 
+    fn execute_write(&mut self) -> Result<Option<Bug>, EngineError> {
+        if !matches!(self.regs[Register::A0 as usize], Value::Concrete(1)) {
+            return not_supported("can not handle other fd than stdout in write syscall");
+        }
+
+        let buffer = if let Value::Concrete(b) = self.regs[Register::A1 as usize] {
+            b
+        } else {
+            return not_supported(
+                "can not handle symbolic or uninitialized buffer address in write syscall",
+            );
+        };
+
+        let size = if let Value::Concrete(s) = self.regs[Register::A2 as usize] {
+            s
+        } else {
+            return not_supported("can not handle symbolic or uinitialized size in write syscall");
+        };
+
+        trace!("write: fd={} buffer={:#x} size={}", 1, buffer, size,);
+
+        if !self.is_in_vaddr_range(buffer) || !self.is_in_vaddr_range(buffer + size) {
+            return not_supported("write syscall failed to access buffer");
```

Comment:
This is an actual bug in the input program, rather than an unsupported feature in the execution engine. Have a look at this: https://github.com/cksystemsgroup/monster/blob/main/src/engine/bug.rs#L25

Reply:
Yeah, I was stumped by the same thing when reading the existing implementation of `read`. But looking at the [error section](https://man7.org/linux/man-pages/man2/write.2.html#ERRORS) of the man page I saw `EFAULT` as one of the handled error cases. So I think writing from an out-of-bounds buffer will just make `write` return -1 and not crash. Should this still be considered a bug in the input program by the engine? WDYT?

However we decide here I feel that we should keep the implementations of `read` and `write` in sync here and either treat this case as a bug in both or neither.

I also double-checked my understanding with the following small C (not C*) snippet, wich indeed prints `res:-1 (errno:14, 'Bad address')` when run:

```
#include <errno.h>
#include <stdio.h>
#include <string.h>
#include <unistd.h>

int main() {
  void* x = (void*) 23;
  int res = write(0, x, 10);
  printf("res:%d (errno:%d, '%s')\n", res, errno, strerror(errno));
}
```

Reply:
Hmm interesting, I did not know that. 

So we have two possibly scenarios. Either `-1` is returned or the engine has found a bug at that point. From a programmers perspective I think I would classify it as a bug. Keep in mind, that undefined behaviour is always a subset of our bug definition. It really comes down to, what we decide to be a bug. `exit(1)` is also a bug in our current view.

Reply:
Done. This is now treated as a bug for both `read` and `write` and will be reported as `Bug::AccessToOutOfRangeAddress` in both cases.

Reply:
nice